tests/farmer_harvester/test_farmer.py

```diff
@@ -619,6 +638,274 @@ def assert_pool_errors_24h() -> None:
+            DummyPoolResponse(False, 500),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "invalid_partials_since_start": 1,
+                    "invalid_partials_24h": [1],
+                }
+            ),
+            id="response_not_ok",
+        ),
+        pytest.param(
+            DummyPoolResponse(True, 200, error_code=uint16(PoolErrorCode.TOO_LATE.value)),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "pool_errors_24h": [{"error_code": uint16(PoolErrorCode.TOO_LATE.value)}],
+                    "stale_partials_since_start": 1,
+                    "stale_partials_24h": [1],
+                }
+            ),
+            id="stale_partial",
+        ),
+        pytest.param(
+            DummyPoolResponse(True, 200, error_code=uint16(PoolErrorCode.PROOF_NOT_GOOD_ENOUGH.value)),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "pool_errors_24h": [{"error_code": uint16(PoolErrorCode.PROOF_NOT_GOOD_ENOUGH.value)}],
+                    "insufficient_partials_since_start": 1,
+                    "insufficient_partials_24h": [1],
+                }
+            ),
+            id="insufficient_partial",
+        ),
+        pytest.param(
+            DummyPoolResponse(True, 200, error_code=uint16(PoolErrorCode.SERVER_EXCEPTION.value)),
+            override_pool_state(
+                {
+                    "points_found_since_start": 1,
+                    "points_found_24h": [1],
+                    "pool_errors_24h": [{"error_code": uint16(PoolErrorCode.SERVER_EXCEPTION.value)}],
+                    "invalid_partials_since_start": 1,
+                    "invalid_partials_24h": [1],
+                }
+            ),
+            id="other_failed_partial",
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_farmer_pool_response(
+    mocker: MockerFixture,
+    harvester_farmer_environment: HarvesterFarmerEnvironment,
+    pool_response_case: DummyPoolResponse,
+    expected_pool_state: Dict[str, Any],
+) -> None:
+    farmer_service, farmer_rpc_client, _, _, _ = harvester_farmer_environment
+    farmer_api = farmer_service._api
+
+    sp, pos, new_pos = create_valid_pos(farmer_api.farmer)
+    assert pos.pool_contract_puzzle_hash is not None
+    p2_singleton_puzzle_hash: bytes32 = pos.pool_contract_puzzle_hash
+
+    assert (
+        verify_and_get_quality_string(
+            pos, DEFAULT_CONSTANTS, sp.challenge_hash, sp.challenge_chain_sp, height=uint32(1)
+        )
+        is not None
+    )
+
+    mock_http_post = mocker.patch("aiohttp.ClientSession.post", return_value=pool_response_case)
+
+    peer: Any = DummyHarvesterPeer(False)
```

Comment:
I think we should `cast()` this explicitly until this is eventually handled properly (outside the scope of this PR).

Reply:
Done